pkg/cmd/cmd.go

```diff
@@ -348,6 +349,62 @@ func setLoggingLevel(cmd *cobra.Command, args []string) {
 			}
 		}
 	}
+
+	logWarningForRemovalOfStaticMasters(cmd, args)
+}
+
+func logWarningForRemovalOfStaticMasters(cmd *cobra.Command, args []string) {
+	// If we're running in a cluster, just don't do anything. There's no reason to flood the build logs, and BDD tests
+	// can depend on parsing specific output.
+	if cluster.IsInCluster() {
+		return
+	}
+
+	shouldWarn := false
+	if commandHasParentName(cmd, "create cluster") {
+		shouldWarn = isStaticMasterInstall(cmd)
+	} else if commandHasParentName(cmd, "install") {
+		shouldWarn = isStaticMasterInstall(cmd)
+	} else {
+		clientFactory := clients.NewFactory()
+		jxClient, ns, err := clientFactory.CreateJXClient()
+
+		// If the jxClient is nil or we got an error, don't bother continuing, something else is wrong enough that it's not worth warning about
+		// removal.
+		if err != nil || jxClient == nil {
+			return
+		}
+
+		// Try to get the dev env.
+		env, err := jxClient.JenkinsV1().Environments(ns).Get(kube.LabelValueDevEnvironment, metav1.GetOptions{})
+
+		// If we get an error or the dev env is nil, just ignore it and move on - something else is wrong enough that it's not worth worrying about
+		// whether we should be warning about removal.
+		if err != nil || env == nil {
+			return
+		}
+
+		teamSettings := &env.Spec.TeamSettings
+		shouldWarn = !teamSettings.IsJenkinsXPipelines()
+	}
+
+	if shouldWarn {
+		cmd.Println("WARNING")
+		cmd.Println("")
+		cmd.Println("Traditional Jenkins masters are being deprecated and removed from all new Jenkins X releases starting in April, 2020.")
+		cmd.Println("We strongly recommend not using this configuration and instead using the default Tekton-based configuration instead for all Jenkins X activity going forward.")
```

Comment:
Log message has 'instead' twice. Possible replacement: "We strongly recommend using the default Tekton-based configuration for all Jenkins X activity going forward. We do not recommend using Jenkins masters."

Reply:
How about `We strongly recommend using the default Tekton-based configuration instead for all Jenkins X activity going forward. We do not recommend using the Jenkins X workflow based around Jenkins masters.`? "We do not recommend using Jenkins masters" is a little too potentially misleading. =)

Reply:
I'm good with `We strongly recommend using the default Tekton-based configuration instead for all Jenkins X activity going forward. We do not recommend using the Jenkins X workflow based around Jenkins masters.`